api/python/provisioner/commands/check.py

```diff
@@ -269,6 +269,42 @@ def make_decision(self, check_result: CheckResult):
         :return:
         """
 
+class DeploymentDecisionMaker(DecisionMaker):
+
+    """Class analyses `CheckResult` structure and will decide
+       to continue or to stop Factory Deployment routine
+    """
+
+    def _check_critical_errors(self, check_result: CheckResult):
+        """
+        Just as example that `DecisionMaker` child classes can threat some
+        errors as critical to raise appropriate Exception and stop command
+        execution
+
+        :param CheckResult check_result: instance with checks results
+        :return:
+        """
+        # TODO: determine list of critical errors which should trigger
+        # Factory Deployment exception
+        pass
+
+    def make_decision(self, check_result: CheckResult):
+        """
+        Make a decision for Factory Deployment
+        based on `CheckResult` analysis
+
+        :param CheckResult check_result: instance with all checks needed for
+                                         to make a decision
+        :return:
+        """
+        if check_result.is_failed:
```

Comment:
looks like that logic is becoming quite useful in multiple subclasses, makes sense to move the common part to the case class and leave only differences in the child classes

Reply:
Comment addressed.